config/value.go

```diff
@@ -476,13 +497,19 @@ func (cv Value) valueStruct(key string, target interface{}) (interface{}, error)
 				val = fieldInfo.DefaultValue
 			}
 			if val != nil {
-				v3, err := convertValue(val, fieldValue.Type())
-				if err != nil {
+				ret, err := convertValue(val, fieldType)
+				// For Unmarshalled text, if convertValue fails, and not returns
```

Comment:
comment should be something like "first try converting primitive values. If nil is returned, it wasn't able to convert to primitive, so try converting as a struct value"